### ISSUE TITLE
Incorrect Momentum Calculation in 16_accel_sgd

### ISSUE THREAD

**Issue by fpgaminer** [fastai/fastbook#174]
I think the momentum calculations in 16_accel_sgd are wrong.

Early in the chapter the formula for weighted average is shown as:

```
weight.avg = beta * weight.avg + (1-beta) * weight.grad
```

That looks correct to me for an exponentially weighted average.  But the actual implementation a little later in the book is:

```
def average_grad(p, mom, grad_avg=None, **kwargs):
    if grad_avg is None: grad_avg = torch.zeros_like(p.grad.data)
    return {'grad_avg': grad_avg*mom + p.grad.data}
```

The last line there is calculating the weight average, but AFAICT its missing the `(1-beta)` term.  Shouldn't it be `return {'grad_avg': grad_avg*mom + p.grad.data*(1.0-mom)}`?

The same thing occurs in the RMSProp implementation.

Is that intentional?  Is the gradient being weighted somewhere else somehow?

As a side note, I noticed with a few runs that training didn't seem to progress much differently with either the versions as they are or with "fixing" them.  Kinda odd, since it seems like an important implementation detail to me...

**Comment by jph00** [comment]
It's intentional - it depends whether damping is used.
